Testing Github Actions results and doing some cross-branch cleanup

akosasante opened the pull request:
**Fixes for Github Actions**
- I forgot to instruct about how to make sure environment variables are available, which is needed by the test step because it actually runs the server and will complain that the cookie/jwt/session secret values are not available
	- created variables in the repo settings:  	https://github.com/Code-the-Dream-School/dd-prac-team4-back/settings/variables/actions
	- Pull the variables into the environment variables in the Github action: https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/28/commits/7e03d14b53e60997823be5d51b0b8b1b99745373
- One test was failing because in a PR subsequent to the test being first created, we added `username` as a required field for creating a user. Fixed by adding a username to the user in the test being created: https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/28/commits/1bbd3df97f4728dcaf6991a4777049ec025cdfc4
- ESLint doesn't by default recognize all the global variables that Jest creates and uses (like `describe`), so I installed the ESLint Jest plugin and set up the eslint rules to have it recognize them so that they won't fail the `npm run lint:check` step: https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/28/commits/9c86db043fc6f7b9ba4a719cd2b8067caacac344
- Updated the steps/scripts so that we are just _checking_ the formatting and linting and not going through the full process of fixing/reformatting in CI since it's a waste since those changes in CI aren't being saved anywhere.
- Ran `npm run lint:fix` and `npm run format` and committed the changes those made + fixed any linting errors it told me about

akosasante committed:
No-op commit to see if actions still fail

akosasante committed:
Add username to test to get it to pass as it is a required field now

akosasante committed:
Echo instructions if linting/formatting fails

akosasante committed:
Ran lint and format to fix remaining issues

NatachaKey commented:
@akosasante Akos, this link returns 404:  https://github.com/Code-the-Dream-School/dd-prac-team4-back/settings/variables/actions how can I access these variables? //but maybe the access is restricted for users

NatachaKey commented:
@akosasante it shows that it has 220 conflicts on package-lock.json- should I resolve them manually or just delete package-lock.json and do npm i or smth like that? 
I just tried to delete some parts of main branch and accept your changes like in this image but the number of conflicts stays the same

UPD: I did npm update to overwrite package-lock.json - i thought it could help to solve that 220 conflicts but now i get 259 😅
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/bedf1c4b-c551-4ed0-9788-6e97ff67a2c3)

NatachaKey commented:
@akosasante just a little question: here:
**_Updated the steps/scripts so that we are just checking the formatting and linting and not going through the full process of fixing/reformatting in CI since it's a waste since those changes in CI aren't being saved anywhere._**

Why don't we just set it up in github actions in our .yml file  to do  'npm run lint:**fix**' instead of
 'npm run lint:**check**'  ?? Probably it could directly fix some issues

NatachaKey commented:
@akosasante during the lastest run - github actions failed again with this PR https://github.com/Code-the-Dream-School/dd-prac-team4-back/actions/runs/5897087497/job/15996128710?pr=15
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/44e25840-9241-47a4-b8f0-86b93cfee501)

Commit:
id npm update

akosasante commented:
> @akosasante Akos, this link returns 404: [Code-the-Dream-School/dd-prac-team4-back/settings/variables/actions](https://github.com/Code-the-Dream-School/dd-prac-team4-back/settings/variables/actions) how can I access these variables? //but maybe the access is restricted for users

Ah yeah :cry:  ok looks like it requires admin access :confused:

akosasante committed:
rebase on main and add fixes

akosasante committed:
fix merge commits and implement lint-staged

generate docs

test

format

Test

test

test with empty commit

fix one more merge conflict

NatachaKey commented:
@akosasante just approved this PR, the only thing it doesn't let me resolve these conflicts, maybe we should delete package-lock.json and add it in new commit?

akosasante commented:
> @akosasante just a little question: here: _**Updated the steps/scripts so that we are just checking the formatting and linting and not going through the full process of fixing/reformatting in CI since it's a waste since those changes in CI aren't being saved anywhere.**_
> 
> Why don't we just set it up in github actions in our .yml file to do 'npm run lint:**fix**' instead of 'npm run lint:**check**' ?? Probably it could directly fix some issues

The github action is running in its own container. So if it does the `npm run lint:fix` the fixes will be applied there in the container. But the syntax erros/misformatted code will still exist on your branch  + local machine + in main once things are merged. It doesn't have a way (simply) to commit any changes it makes. The Github Actions tool is generally meant to report issues / run passive actions rather than permanent ones.

akosasante commented on package.json:
```diff
@@ -43,9 +44,20 @@
     "swagger-autogen": "^2.23.5",
     "swagger-jsdoc": "^6.2.8"
+  },
+  "lint-staged": {
+    "*.js": [
+      "npm run lint:fix",
+      "npm run format"
+    ]
```
@NatachaKey this is similar to what is set up in the frontend repo (together with [husky](https://duncanlew.medium.com/getting-started-with-husky-and-lint-staged-for-pre-commit-hooks-c2764d8c9ae)). It will lint and format code automatically for you whenever you do `git commit`.

akosasante commented on src/middleware/error-handler.js:
```diff
@@ -1,6 +1,6 @@
 const { StatusCodes } = require('http-status-codes');
 
-const errorHandlerMiddleware = (err, req, res, next) => {
+const errorHandlerMiddleware = (err, req, res, _next) => {
```
Marking unused arguments with an underscore prefix; tells ESLint to ignore them / that we are intentionally not using that argument.

akosasante commented:
> @akosasante just approved this PR, the only thing it doesn't let me resolve these conflicts, maybe we should delete package-lock.json and add it in new commit?

In the end I used `npm install --package-lock-only` to resolve this. Included notes in the Knowledge Base: https://codethedreamteam4-2023.atlassian.net/wiki/spaces/PT4/pages/458973/Knowledge+base
